Tella/Scenes/Resources/ViewModel/DownloadedResourcesVM.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  DownloadedResourcesVM.swift
+//  Tella
+//
+//  Created by gus valbuena on 2/5/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class DownloadedResourcesVM : ObservableObject {
+    @Published var appModel: MainAppModel
+    @Published var downloadedResources: [DownloadedResourceCardViewModel] = []
+    
+    let resourceDeleted = PassthroughSubject<Void, Never>()
+    
+    init(mainAppModel: MainAppModel) {
+        self.appModel = mainAppModel
+        fetchDownloadedResources()
+    }
+    
+    func fetchDownloadedResources() {
+        downloadedResources = ResourceService().getDownloadedResources(from: appModel)
+    }
+    
+    func deleteResource(resourceId: String) -> Void {
+        self.appModel.vaultManager.deleteVaultFile(filesIds: [resourceId])
```

Comment:
I think it's better to add the code to tellaData to delete a downloaded resource